kubernetes/utils/create_from_yaml.py

```diff
@@ -54,38 +58,58 @@ def create_from_yaml(
         processing of the request.
         Valid values are: - All: all dry run stages will be processed
     """
+    if path.exists(yaml_file):
+        with open(path.abspath(yaml_file)) as f:
+            content = f.read()
+            try:
+                yaml_file = StringIO(content)
+            except TypeError:
+                yaml_file = StringIO(content.decode('utf-8'))
+
+    yml_document_all = yaml.safe_load_all(yaml_file)
+    # Load all documents from a single YAML file
+    fail_exceptions = []
+
+    for yml_document in yml_document_all:
+        exceptions = create_from_dict(k8s_client, yml_document, verbose,
+                                      **kwargs)
+        if exceptions:
+            fail_exceptions.extend(exceptions)
+
+    if fail_exceptions:
+        raise FailToCreateError(fail_exceptions)
+
+
+def create_from_dict(k8s_client, yml_document, verbose=False, **kwargs):
```

Comment:
if its create_from_dict the argument should not be named `yml_document`
maybe just document, manifest or just data

maybe the whole function should just be called `create_from_data`?
that would also fit the imo reasonable usecase of the function supporting creating objects from the models of this very library.

Reply:
@juliantaylor, the most trivial and important data structure in Python is dict, hence it's very Python to expect something like this. A function named so is easy to understand, already from the name.
When I see a function called `create_from_data` I think "hmm, which data ? Is it pickle? is it some object? Is a protocolbuffer?" So I disagree here.
However, I  do agree about renaming `yml_document` to `data`.

Reply:
you can create objects from all of these as long as it has a mapping interface
the api functions that in the end create the objects also take dictionaries or the models so it would be consistent.
Though there is the `to_dict` function which might work it looks it returns the correct data but interestingly the call_api function does not seem to use it for serialization.
If `to_dict` works then it is probably better to keep the function simple. A comment in the documentation on how to use create with model objects would be helpful then.

Reply:
My latest commit has added a docstring that specifically mentions that data is expected to be a dictionary. Do you think it's lacking? Can you please suggest a better doc-string?

Reply:
It is fine for now, I'll have to do some testing if the `to_dict` function does the right thing (and if not that is what should be fixed).